File: apps/backend/middleware/checkpro.go
package middleware

import (
	"github.com/Tsuzat/Nota/models"
	"github.com/gofiber/fiber/v3"
)

func CheckPro(c fiber.Ctx) error {
	user := c.Locals("user").(*models.User)
	isPro := user.SubscriptionPlan == "pro"
	if !isPro {
		return c.Status(fiber.StatusForbidden).JSON(models.APIError{
			Status: fiber.StatusForbidden,
			Error:  "This feature requires pro plan",
		})
	}
	return c.Next()
}

File: apps/backend-hono/src/app/middlewares/checkpro.ts
import type { Context, Next } from 'hono';

export const proMiddleWare = async (c: Context, next: Next) => {
  const user = c.get('user');
  if (user.subscriptionPlan === 'free') {
    return c.json({ error: 'Pro feature is not available for free users' }, 403);
  }
  await next();
};
